File: src/AssessmentApplication.WebApi/Models/SalesOrderSearchModel.cs
using System;
using AssessmentApplication.Domain.Common;

namespace AssessmentApplication.Models.SalesOrder
{
    public class SalesOrderSearchModel : PagedRequest
    {
        public SalesOrderSearchModel()
        {
            SortBy = "SalesOrderId";
        }

        public string CustomerName { get; set; }

        public DateTime? DueDateEnd { get; set; }

        public DateTime? DueDateStart { get; set; }

        public DateTime? OrderDateEnd { get; set; }

        public DateTime? OrderDateStart { get; set; }

        public DateTime? ShipDateEnd { get; set; }

        public DateTime? ShipDateStart { get; set; }
    }
}

File: src/AssessmentApplication.Angular/src/app/sales-order/models/sales-order-search.model.ts
import { SortDirection } from "@shared/models";
import { PagedRequestModel } from "@shared/models/paged-request.model";

export class SalesOrderSearchModel extends PagedRequestModel {
  public orderDateStart: Date;
  public orderDateEnd: Date;
  public dueDateStart: Date;
  public dueDateEnd: Date;
  public shipDateStart: Date;
  public shipDateEnd: Date;
  public customerName: string;
  [key: string]: Date | number | string | SortDirection | null | undefined;
}
